.readme-partials.yaml

```diff
@@ -606,3 +607,73 @@ body: |-
   ```
 
   A complete example can be found in [`samples/verifyIdToken-iap.js`](https://github.com/googleapis/google-auth-library-nodejs/blob/master/samples/verifyIdToken-iap.js).
+
+  ## Impersonated Credentials Client
+
+  Google Cloud Impersonated credentials used for [Creating short-lived service account credentials](https://cloud.google.com/iam/docs/creating-short-lived-service-account-credentials).
+
+  Provides authentication for applications where local credentials impersonates a remote service account using [IAM Credentials API](https://cloud.google.com/iam/docs/reference/credentials/rest).
+
+  An Impersonated Credentials Client is instantiated with a `sourceClient`. This
+  should be an auth client with the "Service Account Token Creator" role (`roles/iam.serviceAccountTokenCreator`). `sourceClient` is used by the Impersonated
+  Credentials Client to impersonate a target service account with a specified
+  set of scopes.
+
+  ### Sample Usage
+
+  ```javascript
+  const { GoogleAuth, Impersonated } = require('google-auth-library');
+  const { SecretManagerServiceClient } = require('@google-cloud/secret-manager');
+
+  async function main() {
+
+    // Acquire source credentials:
+    const auth = new GoogleAuth();
+    const client = await auth.getClient();
+
+    // Impersonate new credentials:
+    let targetClient = new Impersonated({
+      sourceClient: client,
+      targetPrincipal: "impersonated-account@projectID.iam.gserviceaccount.com",
+      lifetime: 30,
+      delegates: [],
+      targetScopes: ["https://www.googleapis.com/auth/cloud-platform"]
+    });
+
+    // Get impersonated credentials:
+    const authHeaders = await targetClient.getRequestHeaders();
+    // Do something with `authHeaders["Authorization"]`.
+
+    // Use impersonated credentials:
+    try {
+      const url = 'https://www.googleapis.com/storage/v1/b?project=anotherProjectID'
+      const resp = targetClient.request({ url });
+      for (var k in resp.data.items) {
+        console.log(`${resp.data.items[k]['name']}`);
+      }
+    } catch (err) {
+      console.error('Unable to list buckets: ' + error);
```

Comment:
We generally avoid try/catches like this, where we just console.error - we let it throw like you typically would do in production